File: 백준/Bronze/1978. 소수 찾기/소수 찾기.cc
#include <iostream>
using namespace std;

bool is_prime(int n){
    if(n < 2) return false;
    for(int i = 2; i * i <= n; i++){
        if(n % i == 0)return false;
    }
    return true;
}

int main(){
    int N, num;
    cin >> N;
    int sum = 0;

    for(int i = 0; i < N; i++){
        cin >> num;
        if(is_prime(num)) sum ++;
    }
    cout << sum << endl;
    return 0;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    Set<Integer> made = new HashSet<>();
    
    public int solution(String numbers) {
        boolean[] used = new boolean[numbers.length()];
        dfs("", numbers, used);
        
        int count = 0;
        
        for (int n : made) {
            if (isPrime(n)) count++;
        }
        
        return count;
    }
    
    private void dfs(String s, String numbers, boolean[] used) {
        if (!s.isEmpty()){
            made.add(Integer.parseInt(s));
        }
        
        for (int i = 0; i < numbers.length(); i++) {
            if (used[i]) continue;
            
            used[i] = true;
            dfs(s + numbers.charAt(i), numbers, used);
            used[i] = false;
        }
    }
    
    private boolean isPrime(int num) {
        if (num < 2 ) return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        
        return true;
    }
}